core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -569,9 +570,16 @@ private void syncExistingInodeMetadata(LockedInodePath inodePath)
       }
     }
 
-    syncChildren = syncChildren
-        && inode.isDirectory()
-        && mDescendantType != DescendantType.NONE;
+    // Only sync children when
+    // (1) DescendantType.ALL or (2) syncing root of this stream && DescendantType.ONE
```

Comment:
Can we encode this desired behavior in a simple unittest?

Reply:
Gene, there is no unit test for this class `InodeSyncStream` though I think this class should be covered by UT definitely. Do you know if there is any other integration tests covering this part? I am very unfamiliar with this part of source code and it would be a lot to investigate to understand this class and write a UT

Reply:
I think we can use `UfsSyncIntegrationTest` for testing sync behavior.

Reply:
added. PTAL